Macrophages/Macrophages_multiome.py

```diff
@@ -0,0 +1,746 @@
+    )
+    return fig
+    
+# read significantly dysregulated multiome dataset
+df_mean_multi_clustered = pd.read_csv(f'{file_read_path}/df_multi_macrophages_mean_with6clusters.csv',index_col=0)
+
+def get_Kmeans(df, molecular_name):
+    kmeans_value = int(df.loc[molecular_name, 'kmeans'])
+    return kmeans_value
+
+# get co-regulated molecules dataframe
+def select_df_of_same_cluster(df, molecular_name):
+    """
+    Filters the DataFrame based on the kmeans value of the selected molecular name.
+    Parameters:
+    - df: The DataFrame to filter.
+    - molecular_name: The name of the molecule to select.
+    Returns:
+    - filtered_df: The filtered DataFrame.
+    """
+    try:
+        # Check if the molecule exists in the DataFrame
+        if molecular_name in df.index:
+            kmeans_value = df.loc[molecular_name, 'kmeans']
+            filtered_df = df[df['kmeans'] == kmeans_value].iloc[:, :4]  # Select first 4 columns
+            return filtered_df
+        else:
+            return pd.DataFrame()  # Return an empty DataFrame if molecule not found
+    except Exception as e:
+        # General exception handling
+        print(f"An error occurred: {e}")
+        return pd.DataFrame()
+
+# select_df_of_same_cluster(df_mean_multi_clustered, '1/sp|B2RXS4|PLXB2_MOUSE')
+
+def plot_corr_heatmap_selected_molecules(molecular_name):
+    if molecular_name is not None:
+        tar_df = select_df_of_same_cluster(df_mean_multi_clustered, molecular_name).T
+        if not tar_df.empty:
+            # Assuming that the DataFrame has numerical values suitable for a heatmap
+            heatmap_fig = px.imshow(
+                    tar_df,
+                    text_auto=True,                 # Automatically display correlation values on the heatmap
+                    aspect="auto",
+                    color_continuous_scale='RdBu_r',  # Red-Blue color scale for better visualization of positive and negative correlations
+                    zmin=-1.5,                        # Minimum value for the color scale
+                    zmax=1.5,                         # Maximum value for the color scale
+                    labels=dict(x="Molecule", y="Treatment", color="Z-score"),
+                    x=tar_df.columns,
+                    y=tar_df.index,
+                    title=""
+                )
+                # Update layout for better aesthetics
+            heatmap_fig.update_layout(
+                    width=1400,
+                    height=400,
+                    margin=dict(l=100, r=100, t=100, b=100),
+                    xaxis_title=" ",
+                    yaxis_title=" ",
+                    title_x=0.5,  # Center the title
+                    font={'size': 16, 'family': 'Arial'},
+                )
+            heatmap_fig.update_xaxes(showticklabels=False)
+            return heatmap_fig
+    # Return an empty figure if no data is available
+    return {}
+
+def process_data_for_table(molecular_name):

```

Comment:
What kind of processing are you doing? The more specific your function name, the better